cf/data/mixin/deprecations.py

```diff
@@ -69,6 +69,22 @@ def __hash__(self):
             removed_at="5.0.0",
         )
 
+    def __round__(self, *ndigits):
+        """Called to implement the built-in function `round`
+
+        Deprecated at version TODODASK, use method `round` instead.
+
+        x.__round__(*ndigits) <==> round(x, *ndigits)
+      
+        """
+        _DEPRECATION_ERROR_METHOD(
+            self,
+            "__round__",
+            "Use method 'round' instead.",
```

Comment:
I don't think this is an appropriate message, since the `round` method with `cf.Data` ultimately just calls `Data.__round__`? Therefore I suspect the best message to provide in context for the deprecation is something along the lines of that raised when you try to do the equivalent with `numpy`, namely `"cf.Data doesn't define a __round__ method"`?:

```python
>>> n = np.arange(120).reshape(2, 6, 10)
>>> round(n, 5)
Traceback (most recent call last):
  ...
TypeError: type numpy.ndarray doesn't define __round__ method
>>> d = cf.Data(n)
>>> round(d, 5)
Traceback (most recent call last):
  ...
cf.functions.DeprecationError: Data method '__round__' has been deprecated at version TODODASK and is no longer available and will be removed at version 5.0.0. Use method 'round' instead.
```

Reply:
Hi Sadie - sorted: https://github.com/NCAS-CMS/cf-python/pull/370/commits/eff87caa1a99f1356ba10e2bd80d7731ee9febaf

Reply:
Removing the method is indeed a simpler way to produce that message, I suppose! Thanks, marking as resolved.